lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -2246,166 +2254,177 @@ void HdVP2Mesh::_UpdateDrawItem(
         indexBuffer = const_cast<MHWRender::MIndexBuffer*>(sharedBBoxGeom.GetIndexBuffer());
     }
 
-    _delegate->GetVP2ResourceRegistry().EnqueueCommit([stateToCommit,
-                                                       param,
-                                                       primvarInfo,
-                                                       primvars,
-                                                       indexBuffer,
-                                                       isBBoxItem,
-                                                       &sharedBBoxGeom]() {
-        const HdVP2DrawItem::RenderItemData& drawItemData = stateToCommit._renderItemData;
-        MHWRender::MRenderItem*              renderItem = drawItemData._renderItem;
-        if (ARCH_UNLIKELY(!renderItem))
-            return;
+    // We can get an empty stateToCommit when viewport draw modes change. In this case every
+    // rprim is marked dirty to give any stale render items a chance to update. If there are
+    // no stale render items then stateToCommit can be empty!
+    if (!stateToCommit.Empty()) {
+        _delegate->GetVP2ResourceRegistry().EnqueueCommit([stateToCommit,
```

Comment:
The change here is not create the main thread task if there is nothing to do. That buys us a significant performance improvement when we have to do extra updates to get correctness. We can easily create enough empty tasks to cause a "long tail" of serial main thread tasks that do nothing.

The rest of this file is a white space change for the indentation.